File: Set 6/Non-Repeating Character/NRC.py
MAX_CHAR = 26

def NRC(s):
    arr = [0] * MAX_CHAR
    for c in s:
        arr[ord(c) - ord('a')] += 1
    for i in range(len(s)):
        if arr[ord(s[i]) - ord('a')] == 1:
            return s[i]
    return 'NONE'

t = int(input())
for _ in range(t):
    s = str(input())
    print(NRC(s))

File: Set 6/Non-Repeating Character/NRC.cpp
#include <iostream>
#include <vector>
using namespace std;

const int MAX_CHAR = 26;

char NRC(string &s) {
    vector<int> freq(MAX_CHAR, 0);

    for (char c : s) {
        freq[c - 'a']++;
    }

    for (int i = 0; i < s.length(); ++i) {
        if (freq[s[i] - 'a'] == 1)
            return s[i];
    }

    return '$';
}

int main() {
    int t;
    cin >> t;

    while (t--) {
        string s;
        cin >> s;
        cout << NRC(s) << endl;
    }

    return 0;
}
